python/pyspark/sql/pandas/utils.py

```diff
@@ -26,11 +26,13 @@ def require_minimum_pandas_version():
     try:
         import pandas
         have_pandas = True
-    except ImportError:
+    except ImportError as error:
         have_pandas = False
+        raised_error = error
     if not have_pandas:
         raise ImportError("Pandas >= %s must be installed; however, "
-                          "it was not found." % minimum_pandas_version)
+                          "it was not found or an error was thrown by pandas "
+                          "on importing the library." % minimum_pandas_version, raised_error)
```

Comment:
```suggestion
                          "it was not found or an error was thrown by pandas "
                          "it was not found." % minimum_pandas_version) from error
```

Reply:
what about we just leverage exception chaining: https://docs.python.org/3/tutorial/errors.html#exception-chaining?

Reply:
This makes sense for Spark 3.1 - but if we were to backport this to Spark 3.0, 2.4 - we'll need to keep the current syntax as that's the Py2 & Py3-compatible way of doing exception chaining

Happy to move forwards with your suggestion, but keep in mind this will involve us adjusting this implementation in the backports

Reply:
This is just about improvement in exceptions that are usually not ported back in any event. I think it's fine just to land the fix it in master and branch-3.1 only (considering that this is a very safe fix).

Reply:
Gotcha - addressed in 02ba207ccd0c9e89985cfbd4b9f61478e6487b46